src/main/java/com/newrelic/telemetry/opentelemetry/export/MetricPointAdapter.java

```diff
@@ -0,0 +1,179 @@
+package com.newrelic.telemetry.opentelemetry.export;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singleton;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+import com.newrelic.telemetry.Attributes;
+import com.newrelic.telemetry.metrics.Count;
+import com.newrelic.telemetry.metrics.Gauge;
+import com.newrelic.telemetry.metrics.Metric;
+import com.newrelic.telemetry.metrics.Summary;
+import io.opentelemetry.sdk.metrics.data.MetricData.Descriptor;
+import io.opentelemetry.sdk.metrics.data.MetricData.Descriptor.Type;
+import io.opentelemetry.sdk.metrics.data.MetricData.DoublePoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.LongPoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import io.opentelemetry.sdk.metrics.data.MetricData.SummaryPoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.ValueAtPercentile;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MetricPointAdapter {
+
+  // Ideally, we would not have to do this work, and the OTel SDK would be configurable to
+  // make deltas for us automatically.
+  private final Map<Key, DeltaLongCounter> deltaLongCountersByDescriptor = new HashMap<>();
+  private final Map<Key, DeltaDoubleCounter> deltaDoubleCountersByDescriptor = new HashMap<>();
+  private final TimeTracker timeTracker;
+
+  public MetricPointAdapter(TimeTracker timeTracker) {
+    this.timeTracker = timeTracker;
+  }
+
+  Collection<Metric> buildMetricsFromPoint(
+      Descriptor descriptor, Type type, Attributes attributes, Point point) {
+    point.getLabels().forEach(attributes::put);
+    if (point instanceof LongPoint) {
+      return buildLongPointMetrics(descriptor, type, attributes, (LongPoint) point);
+    }
+    if (point instanceof DoublePoint) {
+      return buildDoublePointMetrics(descriptor, type, attributes, (DoublePoint) point);
+    }
+    if (point instanceof SummaryPoint) {
+      return buildSummaryPointMetrics(descriptor, attributes, (SummaryPoint) point);
+    }
+    return emptyList();
```

Comment:
What I wouldn't give for some pattern matching.

Reply:
true true